### ISSUE TITLE
USWDS - Feature Request: Limited spacing of the InputPrefix results in overlapped text

### ISSUE THREAD

**Issue by dakota002** [uswds/uswds#5558]
### Describe the bug

Stylings introduced in https://github.com/uswds/uswds/pull/5063 have broken the visual behavior of the input-prefixes

### Steps to reproduce the bug

I have made a small demo that replicates this bug at https://codesandbox.io/s/funny-artem-s4q8v2?file=/src/App.js

### Expected Behavior

The prefix spacing should not be limited in size to allow for larger words as prefixes, and should apply the same effect to spans or other non-input fields. 

This was the previous functionality in https://github.com/uswds/uswds/tree/v3.1.0, which looked like this:
![image](https://github.com/uswds/uswds/assets/11023698/7e112eb3-522e-4b19-b49d-d6e7a1471eef)

### Screenshots

An example from gcn.nasa.gov. This screenshot shows the overlapping text:
![image](https://github.com/uswds/uswds/assets/11023698/120d9183-4364-4d70-acf9-3fe1606b7db0)

In this screenshot, I have disabled the `position: absolute` styling to show the text that was covered.  ('From' and 'Subject') 
![image](https://github.com/uswds/uswds/assets/11023698/bde8d92b-bcf7-4d7f-b57b-9fa6bbe3a6e7)

### System setup

USWDS version 3.6.0

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by lpsinger** [comment]
I wonder if there is a workaround that we can apply to our custom SCSS so that we can keep using the latest version of USWDS until this is fixed.

**Comment by mahoneycm** [comment]
Confirming I'm able to reproduce the bug. 

Looks like it's coming from us setting the padding to be equal to the icon width:
https://github.com/uswds/uswds/blob/8f92ea70fd8d1fd5d6ca81b5883ceb7bf50fb936/packages/usa-input-prefix-suffix/src/styles/_usa-input-prefix-suffix.scss#L80-L83

We may want to reconsider how this padding is measured while testing for disabled and focus styles.

**Comment by mahoneycm** [comment]
Hey there @lpsinger 

We've added it to our `3.9.0` milestone so it's on our roadmap, but we'll need to get through `3.8.0` first. Are you able to add custom styles as a workaround in the meantime?

**Comment by lpsinger** [comment]
Thanks!

@dakota002, what do you think, is a downstream workaround possible?

**Comment by amyleadem** [comment]
This appears to be two issues:
1. The input prefix suffix appears to be designed to only allow icons or short strings of text
2. There is a feature request for the prefix/suffix to work on non-input elements, like spans

## Next steps
The engineering team should confirm:
- That longer text strings (like "subject" or "ounces") break the input prefix/suffix component
- Confirm if this code sample uses:
    - Valid HTML
    - Follows USWDS guidelines for forms

If this issue is confirmed, we should add the "triage" label and update the weight and complexity (if needed). 

---

Additionally, the usability and accessibility teams should assess that this feature request meets USWDS standards:
- Is it a valid use case to allow for longer strings of text (like "subject") in the input prefix/suffix?
- Is it a valid use case to allow prefix/suffix on non-input elements, like spans?
- When should the content in a prefix/suffix content be a label instead?

**Comment by mahoneycm** [comment]
Hey there @dakota002,

Is the [Code Sandbox Demo](https://codesandbox.io/s/funny-artem-s4q8v2?file=/src/App.js) mentioned in the issue description still available? I tried accessing it to better understand your use case but I was unable to see it.

I'm particularly interested in the HTML structure!

![Image](https://github.com/user-attachments/assets/a5889fb9-f8a3-4ce6-b4ec-2acf8adb7872)

**Comment by dakota002** [comment]
Hey @mahoneycm, 

Sorry, I don't believe it is. I may have made it with a preview that was unassociated with my account at the time. I'll see if I can replicate it and post a refreshed link here in a bit

**Comment by dakota002** [comment]
@mahoneycm Here is a fairly barebones example: https://codesandbox.io/p/devbox/nzdrmz

If you start typing in the TextInput field you will see the entered text overlap with the prefix

**Comment by mahoneycm** [comment]
@dakota002 Much appreciated! When discussing this feature, we're questioning how this would live alongside semantic form markup. In this example, we'd expect a `<label>` element associated with the form input to provide context. 

With this suggestion, are you hoping to use the prefix as a label? We'll likely want to get some UX and accessibility input on this implementation.

**Comment by dakota002** [comment]
> With this suggestion, are you hoping to use the prefix as a label? We'll likely want to get some UX and accessibility input on this implementation.

That is correct, we are using it as if the prefix was the label. Ideally so that we could have inline labels

**Comment by mahoneycm** [comment]
@dakota002 one issue with this approach is in its current implementation, this goes against our [accessibility guidance](https://designsystem.digital.gov/components/input-prefix-suffix/#accessibility-guidance). The prefixes are hidden from screen readers and therefore skipped by screen readers. 

In order to scope the issue a bit, I believe we should focus this feature request on the spacing issue it originally mentioned. If we want to explore the user pattern of using labels within prefix, we should track that in a separate feature request because it will require some additional considerations for this pattern.

So with a focus on the spacing request, I took a look into some other realistic use cases of input prefix/suffix. 

## Prefix

A common use case for a longer prefix was pre-populating some URL paths.

- [Bootstrap Vanity URL example](https://getbootstrap.com/docs/5.0/forms/input-group/#basic-example)
- [Microsoft Fluent UI example](https://arc.net/l/quote/xsgsgnmi)

Replicating these examples with our component does cause text overlap

![Image](https://github.com/user-attachments/assets/ce0e9048-ac39-4166-9f76-ea4736958f68)

## Suffix

This can also be an issue with longer input suffix. Similar to our `lbs` example, using a longer unit of measurement will cause overlap. Another example could be an email creation field with the domain prepopulated:

![Image](https://github.com/user-attachments/assets/f6bdce06-2fd1-4168-8594-78d92533846e)

## Workarounds

Due to the specificity of the styles we set, workarounds can be challenging as well. I tried adding a padding utility class to the input but the specificity doesn't beat our default styles.

**USWDS Input prefix/suffix input padding**
https://github.com/uswds/uswds/blob/8f92ea70fd8d1fd5d6ca81b5883ceb7bf50fb936/packages/usa-input-prefix-suffix/src/styles/_usa-input-prefix-suffix.scss#L80-L83

**Adding a utility class**
```diff
<input type="text" id="example-input-prefix" 
+  class="usa-input padding-left-2"
>
```

Results:

![Image](https://github.com/user-attachments/assets/f8be93c4-93b6-4dc8-81d3-f0a07bd00412)

---

I'm going to bring this to the team for discussion and we'll come back with an update on our next steps!

Thanks so much for your involvement here

**Comment by amyleadem** [comment]
## Next steps
Explore solutions that allow for padding/spacing to adapt to prefix/suffix content. 

We should be aware of how this change affects the focus outline.

**Comment by dakota002** [comment]
Thanks @mahoneycm, we ended up going the utility class route. Appreciate you all looking into this

**Comment by annepetersen** [comment]
Hi there: this is one of many submitted PRs/issues that have languished here. USWDS is currently understaffed, so I'm closing items that aren’t as high on the priority list right now, including this one, unfortunately.
